pytr/event.py

```diff
@@ -110,9 +139,16 @@ def from_dict(cls, event_dict: Dict[Any, Any]):
 
     @staticmethod
     def _parse_type(event_dict: Dict[Any, Any]) -> Optional[EventType]:
-        event_type: Optional[EventType] = tr_event_type_mapping.get(event_dict.get("eventType", ""), None)
-        if event_dict.get("status", "").lower() == "canceled":
-            event_type = None
+        eventTypeStr = event_dict.get("eventType", "")
+        event_type: Optional[EventType] = tr_event_type_mapping.get(eventTypeStr, None)
+        if event_type is not None:
+            if event_dict.get("status", "").lower() == "canceled":
+                event_type = None
+        else:
+            if eventTypeStr not in events_known_ignored:
+                log.warning(f"Ignoring unknown event {eventTypeStr}")
+                if dump_enabled():
+                    dump(f"Unknown event {eventTypeStr}: {pprint.pformat(event_dict, indent=4)}")
```

Comment:
I think this could be handled with `logging` just as well, don't you think? And formatting the `event_dict` as JSON would make it easier to handle, for example to include as sample data in tests.

```suggestion
                self._log.debug("Unknown event %r: %s", eventTypeStr, json.dumps(event_dict, indent=4)))
```

Where the `--dump-debug-data-to-file` option is handled, we can add a [`FileHandler`](https://docs.python.org/3/library/logging.handlers.html#filehandler) to the root logger.

```py
handler = logging.FileHandler(args.dump_debug_data_to_file)
handler.setLevel(logging.DEBUG)
handler.setFormatter(logging.Formatter('%(asctime)s - %(name)s - %(levelname)s - %(message)s'))
logging.root.addHandler(handler)
```

And maybe shorten the option name to `--debug-logfile`?

Alternatively, we could forego the entire option and add a `--debug` or `--verbose` option to set the log level to `DEBUG` such that these messages show up in the console and people can grab it from there; or alternatively have a `--debug --logfile myfile.log` as a combination to achieve the same effect.

Reply:
OK, I'll mull about this. The reason for the `if dump_enabled():` construct is that pretty printing (or json formatting) the data always is quite expensive.